sympy/core/tests/test_power.py

```diff
@@ -403,8 +403,8 @@ def test_issue_7638():
                                                               Rational(3, 2) + I/2]
     assert sqrt(r**Rational(4, 3)) != r**Rational(2, 3)
     assert sqrt((p + I)**Rational(4, 3)) == (p + I)**Rational(2, 3)
-    assert sqrt((p - p**2*I)**2) == p - p**2*I
-    assert sqrt((p**2*I - p)**2) == p**2*I - p  # XXX ok?
+    #assert sqrt((p - p**2*I)**2) == p - p**2*I
+    #assert sqrt((p**2*I - p)**2) == p**2*I - p  # XXX ok?
```

Comment:
These tests are clearly wrong. The left hand sides are obviously equal but the right hand sides are not.

Reply:
The first case seems correct. Is there a way to keep that and correct the 2nd case?

Reply:
I think it can be done like this:
```diff
diff --git a/sympy/core/power.py b/sympy/core/power.py
index 018ae5d..e46715c 100644
--- a/sympy/core/power.py
+++ b/sympy/core/power.py
@@ -470,6 +470,11 @@ def _n2(e):
                     s = 1  # floor = 0
                 elif re(b).is_extended_nonnegative and (abs(e) < 2) == True:
                     s = 1  # floor = 0
+                elif fuzzy_not(im(b).is_zero) and abs(e) == 2:
+                    if re(b).is_extended_nonnegative:
+                        s = 1
+                    elif re(b).is_negative:
+                        s = -1
                 elif _half(other):
                     s = exp(2*S.Pi*S.ImaginaryUnit*other*floor(
                         S.Half - e*arg(b)/(2*S.Pi)))
```
With that we have:
```python
In [2]: p = symbols('p', positive=True)

In [3]: sqrt((p**2*I - p)**2)
Out[3]: 
     2    
- ⅈ⋅p  + p

In [4]: sqrt((p - p**2*I)**2)
Out[4]: 
     2    
- ⅈ⋅p  + p
```
I'm not sure that it's really a good idea though. This routine already has too many cases and is very complicated for ordinary evaluation. Really it would be better to handle this sort of thing in something like `powsimp`.

Reply:
I don't have strong opinion. We can apply this when others complain.